### PR TITLE
fix(analytics): remove zero values being added in charts

### PR DESCRIPTION
**Pull request by anshbansal**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetChartsResolver.java [hunk 1]
```diff
@@ -88,19 +89,25 @@ private TimeSeriesChart getActiveUsersTimeSeriesChart(
    */
   private List<AnalyticsChart> getProductAnalyticsCharts(Authentication authentication) throws Exception {
     final List<AnalyticsChart> charts = new ArrayList<>();
-    final DateTime now = DateTime.now();
-    final DateTime aWeekAgo = now.minusWeeks(1);
+    final DateTime endOfWeek = DateTime.now()
+            .plusDays(1)
```
Comment (chriscollins3456) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetChartsResolver.java:93]: I assume you add  `plusDays(1)` here in case today is Sunday and we need to get the following sunday? and this is all good if today is Saturday?

File: datahub-web-react/src/app/analyticsDashboard/components/TimeSeriesChart.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ type Props = {
 const MARGIN_SIZE = 40;
 
 function insertBlankAt(ts: number, newLine: Array<NumericDataPoint>) {
-    const dateString = new Date(ts).toString();
+    const dateString = new Date(ts).toISOString();
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/analyticsDashboard/components/TimeSeriesChart.tsx:28]: why this change here?

**Review by chriscollins3456 — COMMENTED** [review]
nice! love the tests. two quick questions and then let's move this forward

**Review by jjoyce0510 — APPROVED** [review]
Okay for now. Let's keep an eye on this obviously

File: datahub-web-react/src/app/analyticsDashboard/components/TimeSeriesChart.tsx (hunk 2 above)
Comment (anshbansal) [reply datahub-web-react/src/app/analyticsDashboard/components/TimeSeriesChart.tsx:28]: For the tests basically. Otherwise it would contain people's timezone in the string and it become impossible to test this out for everyone.

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetChartsResolver.java (hunk 1 above)
Comment (anshbansal) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/analytics/resolver/GetChartsResolver.java:93]: What I wanted was the exact end of Sunday. I initially tried that by setting things as 23:59:59.999 but the rendering on the frontend added one point with zero for the exact end. So decided to make it the exact beginning of next Monday by adding plus 1 days.

File: datahub-web-react/src/app/shared/time/__tests__/timeUtils.test.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { addInterval } from '../timeUtils';
+import { DateInterval } from '../../../../types.generated';
+
+describe('timeUtils', () => {
+    describe('addInterval', () => {
+        it('add date interval works correctly', () => {
+            const input = new Date(1677242304000);
+            const afterAdd = addInterval(1, input, DateInterval.Month);
+            const expected = new Date(1679661504000);
```
Comment (iprentic) [inline datahub-web-react/src/app/shared/time/__tests__/timeUtils.test.tsx:9]: This is failing for me because the result is an hour off in my timezone, I think because the next month in my timezone is an hour shorter due to daylight savings? The result I get is 1679657904000. Is the addInterval function calculating the interval in GMT time?

Comment (anshbansal) [reply datahub-web-react/src/app/shared/time/__tests__/timeUtils.test.tsx:9]: https://github.com/datahub-project/datahub/pull/7484 Will see if we can reliably reproduce the failure in CI. Then we will work on fixing it. This should have been caught in the CI
